keyvi/include/keyvi/dictionary/fsa/state_traverser.h

```diff
@@ -176,6 +195,28 @@ class StateTraverser final {
   const traversal::TraversalPayload<TransitionT> &GetTraversalPayload() const { return stack_.traversal_stack_payload; }
 };
 
+template <>
+inline void StateTraverser<traversal::WeightedTransition>::PreIncrement() {
+  TRACE("preincrement weighted transition specialization");
+  stack_.traversal_stack_payload.min_weight = min_weight_;
```

Comment:
indirection might not be necessary, but this could be set directly in line 217

(the 1st version wrote it differently, but this place is global for the whole traversal)